include/sound/soc-acpi.h

```diff
@@ -201,6 +205,7 @@ struct snd_soc_acpi_mach {
 	const char *board;
 	struct snd_soc_acpi_mach * (*machine_quirk)(void *arg);
 	const void *quirk_data;
+	bool (*machine_check)(void *arg);
```

Comment:
this still sounds strange for first and second read, but the `machine_quirk` which returns `snd_soc_acpi_mach` pretty much covers other options (like `machine_match`), I guess it is just fine.